projects/ngrx.io/content/guide/signals/signal-store/entity-management.md

```diff
@@ -1,3 +1,397 @@
+        [
+          { id: 2, name: 'Car Washing', finished: false },
+          { id: 3, name: 'Cat Feeding', finished: false },
+        ],
+        { collection: 'todo' }
+      )
+    );
+  }
+}
+```
+
+The names of the state properties changed from:
+
+- `ids` -> `todoIds`
+- `entities` -> `todoEntities`
+- `entityMap` -> `todoEntityMap`
+
+All functions that operate on entities require a collection parameter. Those are `add*`, `set*`, `update*`, and `remove*`. They are type-safe because you need to provide the collection to avoid getting a compilation error.
+
+If you have a customized id property, you need to include the `idKey` parameter in the object literal, too:
+
+```typescript
+patchState(
+  this.todoStore,
+  addEntities(
+    [
+      { key: 2, name: 'Car Washing', finished: false },
+      { key: 3, name: 'Cat Feeding', finished: false },
+    ],
+    { idKey: 'key', collection: 'todo' }
+  )
+);
+```
+
+To add multiple entities to a state, execute `withEntities` multiple times:
+
+```typescript
+const Store = signalStore(
+  withEntities({ entity: type<Todo>(), collection: 'todo' }),
+  withEntities({
+    entity: type<User>(),
+    collection: 'user',
+  }),
+  withEntities({ entity: type<Category>(), collection: 'category' })
+);
+```
+
+Try to avoid multiple entity types in one store. It is better to have multiple stores, each with a single entity type.
+
+## Extending and Integrating
+
+You will usually want to persist your entities to a backend. Therefore, you must additionally implement `withMethods` and add the necessary methods.
+
+```typescript
+const TodoStore = signalStore(
+  withEntities<Todo>(),
+  withMethods((store) => {
+    const todoService = inject(TodoService);
+
+    return {
+      async load() {
+        const todos = await todoService.findAll();
+        patchState(store, removeAllEntities());
+        patchState(store, setEntities(todos));
+      },
+
+      async add(name: string) {
+        const todo = await todoService.add(name);
+        patchState(store, addEntity(todo));
+      },
+
+      async remove(id: number) {
+        await todoService.remove(id);
+        patchState(store, removeEntity(id));
+      },
+
+      async setFinished(id: number) {
+        await todoService.setFinished(id);
+        patchState(store, updateEntity({ id, changes: { finished: true } }));
+      },
+      async setUnfinished(id: number) {
+        await todoService.setUnfinished(id);
+        patchState(store, updateEntity({ id, changes: { finished: false } }));
+      },
+    };
+  }),
+  withHooks({ onInit: (store) => store.load() })
+);
+```
+
+If your API follows the same pattern, you can even create your own `with*` function:
+
+```typescript
+interface EntityService<Entity> {
+  findAll(): Promise<Entity[]>;
+
+  add(name: string): Promise<Entity>;
+
+  remove(id: number): Promise<void>;
+}
+
+function withPersistedEntities<Entity extends { id: number }>(EntityService: ProviderToken<EntityService<Entity>>) {
```

Comment:
I'd remove this part about custom CRUD feature for two reasons:

- Custom features will be covered on the dedicated page.
- There are several ways of building generic CRUD features, so I think official docs should be unopinionated here and leave this topic for blog posts.

Reply:
Done